business/istio_validations.go

```diff
@@ -165,13 +169,13 @@ func (in *IstioValidationsService) GetIstioObjectValidations(namespace string, o
 			checkers.GatewayChecker{GatewaysPerNamespace: gatewaysPerNamespace, Namespace: namespace, WorkloadsPerNamespace: workloadsPerNamespace},
 		}
 	case kubernetes.VirtualServices:
```

Comment:
Right now, in the validations we pass from validating objects living in the same namespace to validate object from other namespaces and exported to the current namespace.

The translation of the above sentence into code is that that all the checkers which uses VS, DR, SE now will use the exported VS, DR, SE instead. Because it is not enough with the ones just living in the same namespace.

On the other side, the VS, DR, SE checkers do need two inputs: one is the subject to validate (which remains the same: the ones living in the ns) + the exported ones.
If you'd mixed both inputs you'd end up validating both subjects and exported. Which is not necessary.

So I guess that you'd need to add the exported resources to the SidecarChecker, AuthorizationChecker, PeerAuthnChecker and other checkers which have VS/DR/SE as input.

Reply:
Same pattern applies in the namespace-validation method: https://github.com/kiali/kiali/pull/4185/files#diff-4afe91ed13da4c9e145fdccf0596bee0b6c0fc2e29d3da22f112584428159dd3R113